decision-records/2024-06-04--service-to-stop-while-waiting-for-cert.md

```diff
@@ -0,0 +1,106 @@
+**Date:** 2024-06-04<br/>
+**Author:** @sed-i
+
+## Context and Problem Statement
+When an admin forms a tls relation, there is a time window in which:
+
+1. the juju model tells us we must operate with TLS enabled; however
+2. we haven't received the server cert yet.
+
+As a result we had some code ordering issues (e.g. #414):
+- Should we update the config file with the tls config, which includes path to
+  the cert, before we got the cert, only to perfectly reflect the juju model?
+- Should we wait with the tls config and allow the workload to continue to
+  operate in plain http, in contradiction to the juju model?
+- How should we determine within the charm if "TLS is enabled"?
+
+This problem is common to many charms.
```

Comment:
probably, all charms that support tls. 
Should we mention that this is a gap in the juju model and we're here only because of that, so hopefully at some point this adr will be obsoleted by juju itself?

Reply:
You mean that TLS isn't ubiquitous in juju? Seems far fetched to talk about it.

Reply:
I mean that tls is an ubiquitous problem in juju, which juju should model but doesn't yet

Reply:
but yea, too general perhaps?